Libplanet/Blocks/Block.cs

```diff
@@ -126,21 +152,45 @@ private Block(RawBlock rb)
                 new Nonce(rb.Header.Nonce.ToArray()),
                 rb.Header.Miner.Any() ? new Address(rb.Header.Miner) : (Address?)null,
 #pragma warning disable MEN002 // Line is too long
-                rb.Header.PreviousHash.Any() ? new HashDigest<SHA256>(rb.Header.PreviousHash.ToArray()) : (HashDigest<SHA256>?)null,
+                rb.Header.PreviousHash.Any() ? new HashDigest<SHA256>(rb.Header.PreviousHash) : (HashDigest<SHA256>?)null,
 #pragma warning restore MEN002 // Line is too long
                 DateTimeOffset.ParseExact(
                     rb.Header.Timestamp,
                     BlockHeader.TimestampFormat,
                     CultureInfo.InvariantCulture).ToUniversalTime(),
                 rb.Transactions
                     .Select(tx => Transaction<T>.Deserialize(tx.ToArray()))
-                    .ToList()
-            )
+                    .ToList(),
+#pragma warning disable MEN002 // Line is too long
+                rb.Header.PreEvaluationHash.Any() ? new HashDigest<SHA256>(rb.Header.PreEvaluationHash) : (HashDigest<SHA256>?)null,
+                rb.Header.EvaluationDigest.Any() ? new HashDigest<SHA256>(rb.Header.EvaluationDigest) : (HashDigest<SHA256>?)null)
+#pragma warning restore MEN002 // Line is too long
         {
         }
 
+        /// <summary>
+        /// <see cref="Hash"/> is a Hash derived from a serialized <see cref="Block{T}"/>
+        /// after evaluate actions. It is same as <see cref="PreEvaluationHash"/> when
+        /// <see cref="EvaluationDigest"/> is null.
+        /// <seealso cref="PreEvaluationHash"/>
+        /// <seealso cref="EvaluationDigest"/>
+        /// </summary>
         public HashDigest<SHA256> Hash { get; }
 
+        /// <summary>
+        /// <see cref="PreEvaluationHash"/> is a Hash before evaluate actions.
+        /// Used for comparison with nonce during <see cref="Validate"/>.
+        /// <seealso cref="BlockHeader.Validate"/>
+        /// </summary>
+        public HashDigest<SHA256> PreEvaluationHash { get; }
+
+        /// <summary>
+        /// <see cref="EvaluationDigest"/> is a Hash value derived from a dictionary that values
+        /// changed accounts in the status delta that the block has.
+        /// <seealso cref="BlockChain{T}.MineBlock(Libplanet.Address)"/>
+        /// </summary>
```

Comment:
Follow the consistent docs to the `Block<T>()` constructor's parameter.

`<seealso>` cannot be inside `<summary>` (or any other elements like `<param>`).

```suggestion
        /// The hash derived from the result states of every
        /// <see cref="Transaction{T}.Actions"/> of the block's entire
        /// <see cref="Transactions"/>.</param>
        /// </summary>
        /// <seealso cref="BlockChain{T}.MineBlock(Libplanet.Address)"/>
```